docker.go

```diff
@@ -892,6 +892,13 @@ func (p *DockerProvider) GetNetwork(ctx context.Context, req NetworkRequest) (ty
 
 func (p *DockerProvider) GetGatewayIP(ctx context.Context) (string, error) {
 	// Use a default network as defined in the DockerProvider
+	var err error
+	if p.defaultNetwork == "" {
```

Comment:
I think you can move the error variable inside the block it will be used
```suggestion
	if p.defaultNetwork == "" {
		var err error
```